app/src/main/java/com/example/livedata/LoginViewModel.kt

```diff
@@ -0,0 +1,19 @@
+package com.example.livedata
+
+import android.arch.lifecycle.MutableLiveData
+import android.arch.lifecycle.ViewModel
+import android.view.View
+
+class LoginViewModel : ViewModel() {
+    var emailAddress = MutableLiveData<String>()
```

Comment:
We don't want to expose the mutable versions of our LiveData fields to the view i.e. we don't want the view to be able to update them because that violates the direction of data flow. Use instead:
```
private val mutableEmailAddress = MutableLiveData<String>()
val emailAddress: LiveData<String> = mutableEmailAddress
```
The view can only observe the immutable reference. Only your vm can update the immutable because it is private.

Reply:
Same for all live data fields.

Reply:
looks like russell and dz went back and forth on this https://github.com/IntrepidPursuits/skotlinton-android/pull/33. Gonna go with anton's change to the BaseViewModel